### PR TITLE
Js a ts

### PR DESCRIPTION
**Pull request by JaqKent**
js a ts completo :*

### CODE REVIEW

File: ts/app.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    var element = event.target;
+    var elementJob = element.attributes.job.value;
+    if (elementJob == "complete") {
+        completeToDo(element);
+    }
+    else if (elementJob == "delete") {
+        removeToDo(element);
+    }
+    localStorage.setItem("TODO", JSON.stringify(LIST));
+});
```
Comment (nathan-hyan) [inline ts/app.js]: You can delete this file, all js now will be allocated in `./src`

File: ts/app.ts [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 const clear = document.querySelector(".clear");
 const dateElement = document.getElementById("date");
 const list = document.getElementById("list");
-const input = document.getElementById("input");
+const input = document.getElementById("input") as HTMLInputElement;
 
 //Classes names
 const CHECK = "fa-check-circle";
 const UNCHECK = "fa-circle-thin";
 const LINE_THROUGH = "lineThrough";
```
Comment (nathan-hyan) [inline ts/app.ts]: All constants can be exported to a `constants.ts` and imported here, also use `SNAKE_CASE` for const

File: ts/app.ts [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
 
 //variables
-let LIST, id;
+let LIST: ListInterface[], id: number;
```
Comment (nathan-hyan) [inline ts/app.ts:17]: remove spaces

File: ts/app.ts [hunk 4]
```diff
@@ -22,24 +24,35 @@ if (data) {
   id = 0;
 }
 
-function loadList(array) {
-  array.forEach(function (item) {
+function loadList(
+  array: { name: string; id: number; done: boolean; trash: boolean }[]
```
Comment (nathan-hyan) [inline ts/app.ts]: export interface to it's file

File: ts/app.ts [hunk 5]
```diff
@@ -22,24 +24,35 @@ if (data) {
...
+) {
+  array.forEach(function (item: {
```
Comment (nathan-hyan) [inline ts/app.ts:29]: same here, export interface... check if you can reuse the upper one

File: ts/app.ts [hunk 6]
```diff
@@ -22,24 +24,35 @@ if (data) {
...
+  }) {
     addToDo(item.name, item.id, item.done, item.trash);
   });
 }
 
-clear.addEventListener("click", function () {
+clear!.addEventListener("click", function () {
   localStorage.clear();
   location.reload();
 });
```
Comment (nathan-hyan) [inline ts/app.ts]: for all `addEventListener`: export the functions to an `utils.ts` file and reference it here. Move all event listeners to the bottom of the file so it loads everything first and then applies them to the DOM

File: ts/app.ts [hunk 7]
```diff
@@ -22,24 +24,35 @@ if (data) {
...
 
 //date
-const options = { weekday: "long", month: "short", day: "numeric" };
+const options: TimeOptions = {
+  weekday: "long",
+  month: "short",
+  day: "numeric",
+};
```
Comment (nathan-hyan) [inline ts/app.ts]: export to a constant file

File: ts/app.ts [hunk 8]
```diff
@@ -22,24 +24,35 @@ if (data) {
...
 const today = new Date();
```
Comment (nathan-hyan) [inline ts/app.ts]: same here

**Review by nathan-hyan — COMMENTED** [review]
Pretty good job, some comments

File: ts/app.js (hunk 1 above)
Comment (JaqKent) [reply ts/app.js]: done

File: ts/app.ts (hunk 3 above)
Comment (JaqKent) [reply ts/app.ts:17]: done

File: ts/app.ts (hunk 4 above)
Comment (JaqKent) [reply ts/app.ts]: done

File: ts/app.ts (hunk 2 above)
Comment (JaqKent) [reply ts/app.ts]: done

File: ts/app.ts (hunk 5 above)
Comment (JaqKent) [reply ts/app.ts:29]: done

File: ts/app.ts (hunk 6 above)
Comment (JaqKent) [reply ts/app.ts]: done

File: ts/app.ts (hunk 7 above)
Comment (JaqKent) [reply ts/app.ts]: done

File: ts/app.ts (hunk 8 above)
Comment (JaqKent) [reply ts/app.ts]: done

**Review by nathan-hyan — APPROVED** [review]
🥇

**Pull request merged**